frontend/src/container/CreateAlertV2/EvaluationSettings/utils.tsx

```diff
@@ -0,0 +1,246 @@
+import { generateTimezoneData } from 'components/CustomTimePicker/timezoneUtils';
+import dayjs, { Dayjs } from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
+import { rrulestr } from 'rrule';
+
+import { WEEKDAY_MAP } from './constants';
+import { CumulativeWindowTimeframes, RollingWindowTimeframes } from './types';
+
+// Extend dayjs with timezone plugins
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+export const getEvaluationWindowTypeText = (
+	windowType: 'rolling' | 'cumulative',
+): string => {
+	switch (windowType) {
+		case 'rolling':
+			return 'Rolling';
+		case 'cumulative':
+			return 'Cumulative';
+		default:
+			return 'Rolling';
+	}
+};
+
+export const getCumulativeWindowTimeframeText = (timeframe: string): string => {
+	switch (timeframe) {
+		case CumulativeWindowTimeframes.CURRENT_HOUR:
+			return 'Current hour';
+		case CumulativeWindowTimeframes.CURRENT_DAY:
+			return 'Current day';
+		case CumulativeWindowTimeframes.CURRENT_MONTH:
+			return 'Current month';
+		default:
+			return 'Current hour';
+	}
+};
+
+export const getRollingWindowTimeframeText = (
+	timeframe: RollingWindowTimeframes,
+): string => {
+	switch (timeframe) {
+		case RollingWindowTimeframes.LAST_5_MINUTES:
+			return 'Last 5 minutes';
+		case RollingWindowTimeframes.LAST_10_MINUTES:
+			return 'Last 10 minutes';
+		case RollingWindowTimeframes.LAST_15_MINUTES:
+			return 'Last 15 minutes';
+		case RollingWindowTimeframes.LAST_30_MINUTES:
+			return 'Last 30 minutes';
+		case RollingWindowTimeframes.LAST_1_HOUR:
+			return 'Last 1 hour';
+		case RollingWindowTimeframes.LAST_2_HOURS:
+			return 'Last 2 hours';
+		case RollingWindowTimeframes.LAST_4_HOURS:
+			return 'Last 4 hours';
+		default:
+			return 'Last 5 minutes';
+	}
+};
+
+export const getTimeframeText = (
+	windowType: 'rolling' | 'cumulative',
+	timeframe: string,
+): string => {
+	if (windowType === 'rolling') {
+		return getRollingWindowTimeframeText(timeframe as RollingWindowTimeframes);
+	}
+	return getCumulativeWindowTimeframeText(timeframe);
+};
+
+export function buildAlertScheduleFromRRule(
+	rruleString: string,
+	date: Dayjs | null,
+	startAt: string,
+	maxOccurrences = 10,
+): Date[] | null {
+	try {
+		if (!rruleString) return null;
+
+		// Handle literal \n in string
+		let finalRRuleString = rruleString.replace(/\\n/g, '\n');
```

Comment:
Please help me understand what is this about

Reply:
It's used to handle escaped newline characters. If the user directly inputs the rrule it might not be required. But if they're copy pasting it from somewhere, then I thought it's better to handle it. Please let me know it's not necessary.

Reply:
We can keep it.